src/table.rs

```diff
@@ -169,15 +171,11 @@ impl Table {
 impl<T: Slot> Page<T> {
     #[allow(clippy::uninit_vec)]
     fn new(ingredient: IngredientIndex) -> Self {
-        let mut data = Vec::with_capacity(PAGE_LEN);
-        unsafe {
-            data.set_len(PAGE_LEN);
```

Comment:
Afaik this considered UB as we now have a Vec containing uninitialized values (`UnsafeCell` does not make this sound either, `MaybeUninit` would I believe)